apps/OboeTester/app/src/main/java/com/mobileer/audio_device/AudioDeviceInfoConverter.java

```diff
@@ -62,6 +67,36 @@ public static String toString(AudioDeviceInfo adi){
         sb.append("\nSample Rates: ");
         int[] sampleRates = adi.getSampleRates();
         sb.append(intArrayToString(sampleRates));
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            sb.append("\nAddress: ");
+            sb.append(adi.getAddress());
+        }
+
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.R) {
+            sb.append("\nEncapsulation Metadata Types: ");
+            int[] encapsulationMetadataTypes = adi.getEncapsulationMetadataTypes();
+            sb.append(intArrayToString(encapsulationMetadataTypes));
+        }
+
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.R) {
+            sb.append("\nEncapsulation Modes: ");
+            int[] encapsulationModes = adi.getEncapsulationModes();
+            sb.append(intArrayToString(encapsulationModes));
+        }
+
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.S) {
+            sb.append("\nAudio Descriptors: ");
+            List<AudioDescriptor> audioDescriptors = adi.getAudioDescriptors();
```

Comment:
There is not a `toString` function for AudioDescriptor. The print may be meanless.

Reply:
It is there now. I missed that.